### PR TITLE
add hoverfly

### PR DESCRIPTION
**Pull request by pcarney8**
#### What is this PR About?
add hoverfly to container-quickstarts

#### How do we test this?
follow steps in PR to run in openshift

cc: @redhat-cop/containerize-it

### CODE REVIEW

File: hoverfly/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Hoverfly
+
+A simple container to run [hoverfly](https://docs.hoverfly.io/en/latest/). 
+
+TODO create a template
```
Comment (tylerauerbeck) [inline hoverfly/README.md]: @pcarney8 Is there a reason we're not including the template that we already have in this move? This actually resolves the issue that @makentenza is seeing with having to add `"-listen-on-host=0.0.0.0"` to the Dockerfile.

Comment (pcarney8) [reply hoverfly/README.md]: templates are now in https://github.com/rht-labs/openshift-templates and I hate copy/pasting them everywhere.

File: hoverfly/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Usage
+
+### Running in OpenShift
+
+Build the container and deploy it in openshift:
+
+`$ oc new-app https://github.com/rht-labs/labs-ci-cd --name=hoverfly --context-dir=docker/hoverfly`
```
[the remark this answers is not in the document]
Comment (pcarney8) [reply hoverfly/README.md]: good catch

File: hoverfly/README.md (hunk 1 above)
Comment (pcarney8) [reply hoverfly/README.md]: this is also whole PR is a literal copy/paste/remove from labs-ci-cd

**Comment by pcarney8** [comment]
@tylerauerbeck yes, completely agree. I noticed that as well.

**Comment by pcarney8** [comment]
@makentenza can you take a look at the updates? thanks!

**Comment by makentenza** [comment]
@pcarney8 Just reviewed again:

- The URL included in the README.md file is not working --> https://docs.hoverfly.io/en/latest/pages/reference/api/api.html
- I can see you have created the '.applier' directory, but the instructions to deploy this are not updated in the README.md file
- I can't find the file you are referencing in the applier inventory --> `template: "{{ inventory_dir }}/../../build-docker-generic/.openshift/templates/docker-build-template.yml"`

**Comment by pcarney8** [comment]
@makentenza Readme updated. `docker-build-template.yml` was merged in via the updates to master.

**Comment by makentenza** [comment]
@pcarney8 The README file has markdown syntax errors:

https://github.com/redhat-cop/openshift-applier[OpenShift Applier]

....
ansible-galaxy install -r requirements.yml -p galaxy
....

....
oc login <dev cluster>
ansible-playbook -i .applier/ galaxy/openshift-applier/playbooks/openshift-cluster-seed.yml
....

Also, the `requirements.yml` is pointing to 'v2.0.4' of the applier repo which doesn't work on Fedora 29, please move it to 'v2.0.6'.

After making those changes manually I'm able to test it but getting the following error because `applier` is missing the `-f` flag, so this particular issue is not related to this repo:

```
TASK [openshift-applier : Create OpenShift objects based on template with params for 'Environment Setup : Create Projects'] ******************************************************************************************************************
failed: [localhost] (item={'oc_path': ''}) => {"changed": true, "cmd": "oc process   https://raw.githubusercontent.com/redhat-cop/cluster-lifecycle/v3.9.0/files/projectrequest/template.yml   --param='NAMESPACE=hoverfly' --param='NAMESPACE_DISPLAY_NAME=hoverfly'  --ignore-unknown-parameters | oc create  -f - ", "delta": "0:00:00.144350", "end": "2018-12-18 22:07:10.810174", "failed_when_result": true, "msg": "non-zero return code", "oc_param_file_item": {"oc_path": ""}, "rc": 1, "start": "2018-12-18 22:07:10.665824", "stderr": "error: invalid argument \"https://raw.githubusercontent.com/redhat-cop/cluster-lifecycle/v3.9.0/files/projectrequest/template.yml\"\nerror: no objects passed to create", "stderr_lines": ["error: invalid argument \"https://raw.githubusercontent.com/redhat-cop/cluster-lifecycle/v3.9.0/files/projectrequest/template.yml\"", "error: no objects passed to create"], "stdout": "", "stdout_lines": []}
```

**Comment by makentenza** [comment]
@oybed @pcarney8 not sure why is this merged as this is not working, at least for me. I pointed out the following issue during my review and is not resolved yet from my point of view:

--------------------
Nothing is responding to 8500 or 8888 ports, this is because the application is opening a socket only in localhost:

```
sh-4.2$ netstat -naltup
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:8500          0.0.0.0:*               LISTEN      17/hoverfly
tcp        0      0 127.0.0.1:8888          0.0.0.0:*               LISTEN      17/hoverfly
tcp        0      0 127.0.0.1:49068         127.0.0.1:8888          TIME_WAIT   -
tcp        0      0 127.0.0.1:49072         127.0.0.1:8888          ESTABLISHED 24/hoverctl
tcp        0      0 127.0.0.1:8888          127.0.0.1:49072         ESTABLISHED 17/hoverfly
```

I suggest to remove the s2i part here as is not necessary and modify the container CMD to:

```
CMD ["/usr/bin/hoverfly", "-listen-on-host=0.0.0.0"]
```

**Pull request merged**